main.go

```diff
@@ -69,14 +104,38 @@ func main() {
 	}
```

Comment:
Doing way way too much in main. Break this up

Reply:
I agree, I began to break up some of it with activate and scp but I'll continue to pull more out